### PR TITLE
refactor(seasons): move season tab api requests to vuex

### PR DESCRIPTION
**Pull request by bbogdanov**

### CODE REVIEW

File: components/SeasonTabs.vue
Comment (MrTimscampi) [inline components/SeasonTabs.vue:69]: Any reason for using this instead of something like this (Which is what I usually see used) ?
```typescript
...mapState('tvShows', ['seasons', 'seasonEpisodes'])
```

Comment (bbogdanov) [reply components/SeasonTabs.vue:69]: Yea, type safety

**Comment by bbogdanov** [comment]
#165

**Review by MrTimscampi — APPROVED** [review]
This looks good overall.

I have some reservations on some of the stuff around type safety, but that's due to Vuex limitations so... 

I wonder if using [nuxt-typed-vuex](https://github.com/danielroe/typed-vuex) would allow us to clean it up a bit?

I'll wait for a quick review by @dkanada before merging, to get a second opinion.

**Comment by bbogdanov** [comment]
@MrTimscampi I did look into the `nuxt-typed-vuex`.

 My opinion on this is that we still are going to use magic strings and if something change, we are going to need a search/replace for it and with that we might replace something that is not meant to change. 

I actually did a research before proceeding with this implementation and I'm still not seeing something really helpful out there. The implementation here is also not enough cause for example, we still need to type mutations as string and also specify the action scope as string when we dispatch an action, but I believe is a small step in the right direction.

**Comment by MrTimscampi** [comment]
My main concern is that it looks very "non-standard" and hacky, as well as being less approachable to new contributors.

I think we have to strike a balance between type safety, and making the code easy to read and easy to approach for new people.

File: components/SeasonTabs.vue
Comment (dkanada) [inline components/SeasonTabs.vue:71]: I really like this over the old method.

File: store/tvShows.ts
Comment (dkanada) [inline store/tvShows.ts:52]: I agree with @MrTimscampi about this section, it seems a bit cryptic.

File: store/tvShows.ts
Comment (dkanada) [inline store/tvShows.ts:97]: I'm not sure we need separate actions for successes and failures, especially when they'll probably never get called externally.

File: store/tvShows.ts
Comment (dkanada) [reply store/tvShows.ts:52]: Although stores in general are hard to read, so if Vue 3 will improve things it might be fine for now. Perhaps that library is worth the time investment.

Comment (bbogdanov) [reply store/tvShows.ts:52]: Well it really depends. I've used redux in all famous frameworks and to be honest `vuex` is at the bottom of my list for good redux usage inside a framework. All others provide type safety.

File: store/tvShows.ts
Comment (bbogdanov) [reply store/tvShows.ts:97]: For now. At some point you might want to subscribe to that successful action and execute something additional. A quick example on top of my head is disabling a loading spinner.

Comment (bbogdanov) [reply store/tvShows.ts:97]: For the failure I believe the example is pretty much there in the code

**Comment by bbogdanov** [comment]
@MrTimscampi @dkanada I'm not against continuing with what we are used to for the actions. If you think current approach is too much at this stage I'll return the actions approach from the previous commit and continue with the old way.

**Comment by MrTimscampi** [comment]
> @MrTimscampi @dkanada I'm not against continuing with what we are used to for the actions. If you think current approach is too much at this stage I'll return the actions approach from the previous commit and continue with the old way.

I think, for now, going back is the best course of action. Vuex 4 should make TypeScript support better and we can maybe add nuxt-typed-vuex as a middle ground for now.

It's not ideal, but I think "approachability" is pretty important (it's one of the things I've been trying to focus on) and your idea, while it brings type safety, has the potential to scare off potential new contributors who might not be fully used to Vue, Vuex and the rest of the ecosystem.

File: store/tvShows.ts
Comment (Camc314) [inline store/tvShows.ts:61]: This should be translated.

**Comment by bbogdanov** [comment]
> > @MrTimscampi @dkanada I'm not against continuing with what we are used to for the actions. If you think current approach is too much at this stage I'll return the actions approach from the previous commit and continue with the old way.
> 
> I think, for now, going back is the best course of action. Vuex 4 should make TypeScript support better and we can maybe add nuxt-typed-vuex as a middle ground for now.
> 
> It's not ideal, but I think "approachability" is pretty important (it's one of the things I've been trying to focus on) and your idea, while it brings type safety, has the potential to scare off potential new contributors who might not be fully used to Vue, Vuex and the rest of the ecosystem.

Done!

**Review by MrTimscampi — APPROVED** [review]

**Pull request merged**
